src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -873,6 +844,115 @@ internal int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancella
             return Succeeded;
         }
 
+        private bool CompileAndEmitRefAssembly(Compilation compilation, EmitOptions refEmitOptions, string finalRefPeFilePath,
+            TouchedFileLogger touchedFilesLogger, DiagnosticBag diagnosticBag, CancellationToken cancellationToken)
+        {
+            var refModuleBeingBuilt = compilation.CheckOptionsAndCreateModuleBuilder(
+                diagnosticBag,
+                Arguments.ManifestResources,
+                refEmitOptions,
+                debugEntryPoint: null,
+                sourceLinkStream: null,
+                embeddedTexts: null,
+                testData: null,
+                cancellationToken: cancellationToken);
+
+            bool success = true;
+            if (refModuleBeingBuilt != null)
+            {
+                try
+                {
+                    var diagnostics = DiagnosticBag.GetInstance();
```

Comment:
>diagnostics [](start = 24, length = 11)

It looks like we are leaking the bag. Also, consider adding a comment explaining why it is Ok to drop this diagnostics on the floor.